### PR TITLE
Adds FIPS 180-4 512/t variants

### PR DESCRIPTION
**Pull request by jtmcdole**
512/224 and 512/256 are variants specified in the FIPS 180-4 and
commonly available in other packages.

### CODE REVIEW

**Comment by listepo** [comment]
Any news?

**Comment by jtmcdole** [comment]
I need to resync and put this back up. Was blocked on another CL

**Comment by jtmcdole** [comment]
Updated; but I don't know how to restart the travis check. The errors look to be on their side.

**Comment by listepo** [comment]
@jtmcdole https://stackoverflow.com/questions/17606874/trigger-a-travis-ci-rebuild-without-pushing-a-commit maybe it will help

**Comment by jtmcdole** [comment]
I totally forgot about this branch!

**Comment by listepo** [comment]
@jtmcdole it happens) Do you have the opportunity to complete what you started?

**Comment by listepo** [comment]
Any news? Very important for the implementation of JWT

**Comment by jtmcdole** [comment]
Ah - didn't know someone was depending on it for JWT

**Comment by jtmcdole** [comment]
Waiting on presubmits to run

**Comment by jtmcdole** [comment]
@lrhn or anyone interested in reviewing this?

File: CHANGELOG.md
Comment (lrhn) [inline CHANGELOG.md:1]: I think you can stay on 2.2.0-nullsafety-dev.
The "-dev" suffix means (or should mean) that the changes haven't been published yet, so you're just including more into the same release.

If anyone is picking the version directly from github, they are using a hash, not a version, so they won't be affected.

**Review by lrhn — APPROVED** [review]
LGTM. The comments are bad. They always were, but they still are. Just keep them as-is, and I'll take a look at all of them afterwards.

Do make the new `Sha256224`/`Sha256256` classes extends Hash directly, make them private and type the public instances as `Hash`. That's what the rest of them should have done too.

File: lib/src/sha512.dart
Comment (jtmcdole) [reply lib/src/sha512.dart:27]: I'm going to convert the leading ones to just the class names - the `[Specification][Link]` does work in the description.

Comment (jtmcdole) [reply lib/src/sha512.dart:27]: Done

File: lib/src/sha512.dart
Comment (jtmcdole) [reply lib/src/sha512.dart:30]: Done

File: lib/src/sha512.dart
Comment (jtmcdole) [reply lib/src/sha512.dart:83]: A lot of the style / documentation comes from a year or more ago; it was copied from whomever wrote the Sha256 class (which has a private constructor and the `newInstance`  "factory". This are **not** private classes as they are exported by `crypto.dart` and down-stream packages could be using them already.

So I can extend `Hash` just fine, but not make the class private.

File: CHANGELOG.md
Comment (jtmcdole) [reply CHANGELOG.md:1]: Done.

**Comment by jtmcdole** [comment]
Fyi: I do not have merge rights to this repository :)

File: lib/src/sha512.dart
Comment (lrhn) [reply lib/src/sha512.dart:87]: You can make the *new* classes private without anyone being affected. I agree that you can't touch the existing ones.

Comment (jtmcdole) [reply lib/src/sha512.dart:87]: Is the preferred way to make the class private by adding a `_` to the front of the name, or by  using a `hide` on the export?

Comment (jtmcdole) [reply lib/src/sha512.dart:83]: Ok, I've removed the oddball `newInstance`, changed to `_` prefix, and updated the documentation. Weeeeeee.

**Comment by lrhn** [comment]
LGTM! Thanks.

**Pull request merged**
